### ISSUE TITLE
How to start and run command inside created container?

### ISSUE THREAD

**Issue by amihaiemil** [spotify/docker-client#937]
The following piece of code:

```java
final String[] command = {"sh", "-c", scripts};
final ExecCreation execCreation = this.client.execCreate(
    containerId,
    command,
    DockerClient.ExecCreateParam.attachStdout(),
    DockerClient.ExecCreateParam.attachStderr()
);
```

throws 

```java
java.lang.IllegalStateException: Container <<containerId>> is not running.
```

I do not understand why this happens. ``containerId`` is the id of a previously created (not running) container. If I call ``dockerClient.startContainer(containerId)`` before all this, it still won't work because the container closes right away if it has no running process inside.

Please, explain what am I doing wrong, I can't find any explanation in this repo, other than the poor examples from the README and User Manual.

Many thanks,
Mihai

**Comment by amihaiemil** [comment]
@davidxia Maybe you can answer my question? I noticed you are one of the maintainers.

**Comment by davidxia** [comment]
What version of docker-client are you using? I think the container needs to be running for you to attach to stdout and stderr. https://docs.docker.com/engine/reference/commandline/attach/ Not 100% certain though.

**Comment by amihaiemil** [comment]
@davidxia I am using version ``8.9.2``. Thanks for the hint, I'll try it out later today.

**Comment by johnflavin** [comment]
`execCreate` is not an analog to `docker run <image> <command>...`. `execCreate` is an analog to `docker exec <containerId> <command>`, where you execute some command in a container that is already running some long-lived process.

It appears to me that you want to create + start a container with a command. This is the analog of `docker run <image> <command>`.

```java
final String image = "my_cool_image";
final String[] command = {"sh", "-c", scripts};
final ContainerConfig containerConfig = ContainerConfig.builder()
    .image(image)
    .cmd(command)
    // Probably other configuration
    .build();
final ContainerCreation containerCreation = client.createContainer(containerConfig);
client.startContainer(containerCreation.id());
```

**Comment by amihaiemil** [comment]
@johnflavin Yes, I would like to create a container based on an image (say ``ubuntu``), an environment, start it, run some commands inside it and exit. I have to be able to see the logs, and ideally, I should be notified if any of the commands failed. The container should also closed and removed after the commands are executed, successed or failured or if the commands take too long, close after a period of time.

I will try out both your options and come back, thanks!
